### ISSUE TITLE
Static linking with cuBLAS?

### ISSUE THREAD

**Issue by panosk** [OpenNMT/CTranslate2#662]
I'm trying to build ctranslate2 with CUDA support but in the end the executable asks for the monstrous `cublas64_11.dll` and `cublasLt64_11.dll` which are nearly 400MB. Can this be avoided? 
Thanks!

**Comment by guillaumekln** [comment]
You can try hardcoding the path to the cuBLAS static libraries and see if it makes a difference. Search and replace `${CUDA_CUBLAS_LIBRARIES}` in the main `CMakeLists.txt`.

Let me know how it impacts the final size.

**Comment by panosk** [comment]
Thanks, already did that but nothing changes, the dlls are still needed.

I think `cublas_static.lib` is not available for Windows. In my lib directory I can only see `cudart_static.lib`, alongside with `cudart.lib`, but only `cublas.lib` without a corresponding `_static` version. I will keep searching for a while to figure out what's going on.

**Comment by guillaumekln** [comment]
In any case, last time I tried static linking against cuBLAS on Linux the resulting binary was still very big.

A workaround is to compile the project with `-DCUDA_DYNAMIC_LOADING=ON` so that the cuBLAS library is dynamically loaded at runtime. You can then distribute the cuBLAS library as an optional component.

**Comment by panosk** [comment]
That's an option too, thanks. As it seems, for Windows the *.lib files provided with the CUDA SDK are import libraries and not complete static libraries, that's why the dlls are needed. 

The same is true for oneAPI's dnnl.lib that is installed with Intel's installer. After I noticed your recent commit for static linking with dnnl, I compiled dnnl from sources with the `STATIC` option and I got rid of the huge dnnl.dll, adding just ~10MB in the executable size. It seems I can't do that for cuBLAS so I should live with that.
